Add `prefer-event-target` rule (#1792)
sindresorhus/eslint-plugin-unicorn: docs/rules/prefer-event-target.md

# Prefer `EventTarget` over `EventEmitter`

<!-- Do not manually modify RULE_NOTICE part. Run: `npm run generate-rule-notices` -->
<!-- RULE_NOTICE -->
✅ *This rule is part of the [recommended](https://github.com/sindresorhus/eslint-plugin-unicorn#recommended-config) config.*
<!-- /RULE_NOTICE -->

While [EventEmitter](https://nodejs.org/api/events.html#class-eventemitter) is only available in Node.js, [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget) is also available in *Deno* and browsers.

This rule reduces the bundle size and makes your code more cross-platform friendly.

See the [differences](https://nodejs.org/api/events.html#eventtarget-and-event-api) between `EventEmitter` and `EventTarget`.

## Fail

```js

> import EventEmitter from 'node:event';
> 
> class Foo extends EventEmitter {
> 
> }

Reviewer: ```suggestion
import {EventEmitter} from 'node:event';

class Foo extends EventEmitter {}
```

```

```js

> const emitter = new EventEmitter;

Reviewer: ```suggestion
const emitter = new EventEmitter();
```

```

## Pass

```js

> class Foo extends EventTarget {
> 
> }

Reviewer: ```suggestion
class Foo extends EventTarget {}
```

```

```js

> const target = new EventTarget;

Reviewer: ```suggestion
const target = new EventTarget();
```

> # Prefer EventTarget instead of EventEmitter

Reviewer 2: ```suggestion
# Prefer `EventTarget` over `EventEmitter`
```

<!-- Do not manually modify RULE_NOTICE part. Run: `npm run generate-rule-notices` -->
<!-- RULE_NOTICE -->
✅ *This rule is part of the [recommended](https://github.com/sindresorhus/eslint-plugin-unicorn#recommended-config) config.*

<!-- /RULE_NOTICE -->

While `EventEmitter` could be used in only Node.js, `EventTarget` exist in *Deno* and *Browser* too.

> This rule could potentially reduce the bundle size, and make your code more cross-platform friendly.

Reviewer 2: ```suggestion
This rule could potentially reduce the bundle size and make your code more cross-platform friendly.
```

Reviewer 2: It wouldn't potentially reduce bundle size, it would.

> You can check their [differences](https://nodejs.org/api/events.html#eventtarget-and-event-api) between `EventEmitter` and `EventTarget`.

Reviewer 2: ```suggestion
See the [differences](https://nodejs.org/api/events.html#eventtarget-and-event-api) between `EventEmitter` and `EventTarget`.
```

## Fail

```js

> import EventEmitter from 'node:event'

Reviewer 2: ```suggestion
import EventEmitter from 'node:event';
```
